spec/lib/vanagon/project_spec.rb

```diff
@@ -308,6 +408,31 @@ class Project
     end
   end
 
+  describe '#publish_yaml_settings' do
+    let(:platform_name) { 'aix-7.2-ppc' }
+    let(:platform) { Vanagon::Platform.new(platform_name)}
+    let(:publishing_project_block) {
+      "project 'settings-test' do |proj|
+        proj.setting(:key, 'value')
+        proj.version 'version'
+        proj.publish_yaml_settings
+       end"
+    }
+
+    it 'writes project settings as yaml and a sha1sum for the settings to the output directory' do
+      project = Vanagon::Project::DSL.new('test-project', platform_name, {})
+      project.instance_eval publishing_project_block
+      yaml_file = double('yaml_file')
+      sha1_file = double('sha1_file')
+
+      expect(File).to receive(:open).with("output/test-project-version.#{platform_name}.settings.yaml", "w").and_yield(yaml_file)
+      expect(File).to receive(:open).with("output/test-project-version.#{platform_name}.settings.yaml.sha1", "w").and_yield(sha1_file)
+      expect(yaml_file).to receive(:write).with({key: 'value'}.to_yaml)
+      expect(sha1_file).to receive(:write)
+      project._project.publish_yaml_settings(platform)
+    end
+  end
+
```

Comment:
Happy paths testing look pretty solid. Consider more unhappy path options, such as an existing YAML file with missing SHA1 (vice-versa is possible, but not really likely). Are more merge scenarios needed?